warehouse/oidc/models/github.py

```diff
@@ -125,6 +128,56 @@ class GitHubPublisherMixin:
         "enterprise",
     }
 
+    @staticmethod
+    def __lookup_all__(klass, signed_claims: SignedClaims) -> Query | None:
```

Comment:
Flagging for review: this exemplifies the "nastiness" I was talking about: this is conceptually a `classmethod`, but it's registered as a `staticmethod` because it can't be made callable until its surrounding class is finalized.